### DISCUSSION TITLE
Is it possible to react with Pyscript to a long press at a switch?

### DISCUSSION THREAD

**Discussion by wrfz** [custom-components/pyscript#529] [Q&A] [answered]
Is it possible to react with Pyscript to a **long press** on a switch?
The HA atomatization allows to specify a trigger "key long pressed".
Haven't found a way with pyscript yet unfortunately.

Some additional info:
- The Shelly uses the standard Shelly integration.
- ColoT is active.
- MQTT is off.

Any help would be appreciated!

**Comment by FenycFur** [comment, marked as answer]
You can but you need to know what events your device is firing first. What's your event provider? Is it Zigbee HomeAssistant (zha)?

In my case I was able to look at the event types coming from zha_event on this endpoint, [http://homeassistant.local:8123/developer-tools/event](http://homeassistant.local:8123/developer-tools/event) and have my pyscript service fire on every event, then filter out the events I don't care about, and action on those I do care about.
This image shows the event fired from an Ikea Tradfri E1810 5 button remote.
I correlate remote to zha_event event via the **device_id** field, and the type of event (button up/down/press/hold) via a combination of **command+args** fields.
![image](https://github.com/custom-components/pyscript/assets/23361242/cb4a9321-7766-4db8-81d4-a627844de573)

**Reply by wrfz** [reply]
Thanks @JacobLChrzanowski !

The hint to use **development tools->events** has brought me further!
I concentrated all the time on States and ignored Events completely.
This realization works perfectly for my Shelly:

```
@event_trigger("shelly.click", "device == 'shelly1-XXXXXXXXXXXX' and click_type == 'long'")
def corridor_switch_longpress(**kwargs):
    log.info("corridor_switch_longpress")
    cover_open("all")
```

**Reply by FenycFur** [reply]
You got it!

**Reply by FenycFur** [reply]
I do something similar in my pyscript repo. Super WIP.
https://github.com/JacobLChrzanowski/HomeAssistant/blob/cbce3b1d5703c750a2df7261a7eb427a22f28b7d/CONFIG/pyscript/example.py#L383
